frontend/src/employee-frontend/components/unit/unit-reservations/StaffAttendanceTable.tsx

```diff
@@ -127,19 +135,14 @@ const AttendanceRow = React.memo(function AttendanceRow({
           partialRow={false}
           rowIndex={rowIndex}
         >
-          {attendances?.length > 0 ? (
-            attendances.map((attendance) => (
+          {attendancesForDay(day.date.toSystemTzDate())?.length > 0 ? (
+            attendancesForDay(day.date.toSystemTzDate()).map((attendance) => (
```

Comment:
Is the `.length > 0` check needed? Now this filters and sorts the attendances twice just to see how many there are, and then again to map over them.

It might be even better to precompute the `(operationalDay, attendances)` data once and then iterate over it in the `.map()` call.

Reply:
Due to the `useCallback` it should do the actual filtering only once and use the memoized result on the second invocation. When merging this into #2252 it will likely be exactly what you suggest, i.e. precomputing and then iterating over the result.

Reply:
Maybe I'll just do this fix in #2252 and close this?